packages/bundle-analyzer/src/options.ts

```diff
@@ -11,17 +11,40 @@ export interface BundleAnalyzerOptions {
    * @returns {Promise<Output>} A promise that resolves to the customized output.
    */
   beforeReportUpload?: (original: Output) => Promise<Output>;
+
+  /**
+   * Patterns to ignore certain files or directories in the analysis. The patterns follow the
+   * standard glob pattern syntax. By default, it returns an empty list.
+   */
+  ignorePatterns?: string[];
+
+  /**
+   * A pattern to normalize asset names (e.g., for hashing). By default, it returns an empty string,
+   * which will replace anything "hashlike" with "*".
+   */
+  normalizeAssetsPattern?: string;
+
+  /**
+   * Additional directories to include in the build analysis. By default, it returns an empty list.
+   */
+  additionalBuildDirectories?: string[];
```

Comment:
What do you think about changing the API of `createAndUploadReport` to accept an array? I feel like that would be easier to use than to have both `buildDirectoryPath` as the first argument, and then `additionalBuildDirectories` on top of that.

We could also make `buildDirectoryPath` of `createAndUploadReport` be able to take a glob string, so that it still is a string argument, but can be used for array-like queries.

Reply:
Makes sense! My original thought there was that the default usage in most cases would expect a single build directory. And then if people are doing something off the usual path or "fancier" could supply the additional paths optionally.
I went ahead and pulled them together to one array input to reduce the complexity. 
Glob string is an interesting idea - I think I can add it on top of the array in the future if it seems users want to go in that direction